service/raft/raft_group0.hh

```diff
@@ -218,18 +218,17 @@ public:
     // `wait_for_raft` must've also been called earlier and returned `true`.
     future<> become_nonvoter(abort_source& as, std::optional<raft_timeout> timeout = std::nullopt);
```

Comment:
Wondering whether we could remove this one as well - we could pass the "my_id" explicitly and handle the current node the same way as any other node.

On the other side, this would provide some kind of encapsulation in case the current id wouldn't be exposed ... however it currently is (and is even being checked against in the topology coordinator).

Reply:
On the contrary. I am thinking if it worth encapsulating `set_voters_status(false)` into `make_nonvoter()` and 
`set_voters_status(true)` to `make_voter()`. Easier to read code where it is used IMO.

Reply:
Yeah, the idea was to simplify the interface by having less functions ... I was also thinking about creating an enum for voter vs non-voter (instead of just using true/false), do you think it would be enough to make the code cleaner?

But I agree make_nonvoter / make_voter is good for readability, it's just it scales relatively poorly.

As for the `become_nonvoter` vs passing the id explicitly, one thing in particular is that it will again complicate the code on the topology coordinator side where a wrapper (that will handle the voter/non-voter state transitions) will again have to distinguish between these (if the raft voters api wouldn't distinguish these it might sinplify the handling on the topology coordinator side).

Reply:
Maybe I can also add the wrapper (the initial version that just forwards the calls) to this PR as well to see how it would be managed

Reply:
> Yeah, the idea was to simplify the interface by having less functions ... I was also thinking about creating an enum for voter vs non-voter (instead of just using true/false), do you think it would be enough to make the code cleaner?

For such cases we have `bool_class`

> 
> But I agree that make_nonvoter / make_voter is good for readability, it's just it scales relatively poorly.
> 
> As for the `become_nonvoter` vs passing the id explicitly, one thing in particular is that it will again complicate the code on the topology coordinator side where a wrapper (that will handle the voter/non-voter state transitions) will again have to distinguish between these (if the raft voters api wouldn't distinguish these it might sinplify the handling on the topology coordinator side).

`make_nonvoter / make_voter` should be able to handle local node.

Reply:
> `make_nonvoter / make_voter` should be able to handle local node.

Yeah, that's why I was thinking if we actually need the `become_nonvoter` variant. Currently the documentation of `make_nonvoter` explicitly states that:
```
Make the given server, other than us, a non-voter in group 0.
```

so it "shouldn't" be used for the local node. However there is no check or anything inside that would prevent it indeed.

Reply:
Looks like a documentation thing. They both call the same function in the end.

Reply:
> Yeah, the idea was to simplify the interface by having less functions ... I was also thinking about creating an enum for voter vs non-voter (instead of just using true/false), do you think it would be enough to make the code cleaner?

I like this idea. would had made the code more explicit.

Reply:
Added a typed bool using the `bool_class`